qiskit/providers/ibmq/api/clients/runtime.py

```diff
@@ -106,6 +107,23 @@ def program_get_data(self, program_id: str) -> Dict:
         """
         return self.api.program(program_id).get_data()
 
+    def program_set_visibility(self, program_id: str, visibility: str) -> None:
+        """Sets a program's visibility to public.
+
+        Args:
+            program_id: Program ID.
+            visibility: the visibility of the program (public/private)
+
+        Raises:
+            IBMQInputValueError: if visibility is invalid (valid options are private and public)
+
+        """
+        if visibility == 'private':
+            self.api.program(program_id).make_private()
+        if visibility == 'public':
+            self.api.program(program_id).make_public()
+        raise IBMQInputValueError('Invalid program visibility (%s) specified!' % visibility)
```

Comment:
The validation should happen in the higher level method (the one called by the user) to make the traceback stack cleaner. Also the message should tell the user what valid values are.

And since there is no return statement, wouldn't the raise always happen?